impl/client/src/main/javascript/web/prompting/builders/TextInputBuilder.js

```diff
@@ -49,76 +49,110 @@
+        } catch(e) {
+          return DojoNumber.format(val, {locale: "en"});
         }
-        var widget = this.base(args);
-        var name = widget.name + "-input";
-        $.extend(widget, {
-          name: name,
-          type: 'TextInputComponent',
-          preChange: function(){
-            var val = $("#"+this.name).attr('value');
-            this.dashboard.setParameter(this.parameter, parseNumber(val));
-          },
-          postExecution: function(){
-            this.base();
+      }
+      var widget = this.base(args);
+      var name = widget.name + "-input";
+      $.extend(widget, {
+        name: name,
+        type: "TextInputComponent",
+        preChange: function() {
+          var val = $("#" + this.name).attr("value");
+          this.dashboard.setParameter(this.parameter, parseNumber(val));
+          currentDashboard = this.dashboard;
+        },
+        postExecution: function() {
+          this.base();
 
-            var initialValue;
-            $.each(this.param.values, function(i, v) {
-              if (v.selected) {
-                initialValue = this.formatter ? this.formatter.format(this.transportFormatter.parse(v.value)) : v.value;
+          var initialValue;
+          $.each(this.param.values, function(i, v) {
+            if(v.selected) {
+              initialValue = this.formatter ? this.formatter.format(this.transportFormatter.parse(v.value)) : v.value;
 
-                try {
-                  if (isNaN(v.value) || Math.abs(v.value) == Infinity) {
-                    var valueParsed = null;
+              try {
+                if(isNaN(v.value) || Math.abs(v.value) == Infinity) {
+                  var valueParsed = null;
+                } else {
+                  if(Util.isNumberType(v.type)) {
+                    valueParsed = formatNumber(v.value);
                   } else {
-                    if (Util.isNumberType(v.type)) {
-                      valueParsed = formatNumber(v.value);
-                    } else {
-                      valueParsed = v.value;
-                    }
+                    valueParsed = v.value;
                   }
-                } catch (e) {
-                  valueParsed = v.value;
                 }
+              } catch(e) {
+                valueParsed = v.value;
+              }
 
-                if (valueParsed != null) {
-                  initialValue = v.label = v.value = valueParsed;
-                }
+              if(valueParsed != null) {
+                initialValue = v.label = v.value = valueParsed;
               }
-            }.bind(this));
+            }
+          }.bind(this));
 
-            $("#"+this.name).attr('value', initialValue);
+          $("#" + this.name).val(initialValue);
+          currentDashboard = undefined;
+        },
+        refreshOnEveryKeyUp: args.param.attributes.refreshOnEveryKeyUp,
+        clearButton: args.param.attributes.clearButton,
```

Comment:
Can we rename `clearButton` for example to `clearIcon` because originally it's not a button.